asserts/database.go

```diff
@@ -461,3 +483,45 @@ var DefaultCheckers = []Checker{
 	CheckTimestampVsSigningKeyValidity,
 	CheckCrossConsistency,
 }
+
+// no-authority assertion checkers
+
+// NoAuthorityCheckSignature checks that the signature is valid.
+func NoAuthorityCheckSignature(assert Assertion, roDB RODatabase) error {
+	selfSigned, ok := assert.(selfSignedAssertion)
+	if !ok {
+		return fmt.Errorf("cannot check non-self-signed assertion type %q", assert.Type().Name)
```

Comment:
I wonder if this is really a panic? It feels like a fundamental error inside this package rather than something that might happen at runtime.